scripts/calculator.js

```diff
@@ -21,12 +21,31 @@ const addNumber = (num) => {
     currentStep = 'firstNumber'
   } else {
     currentStep = 'secondNumber'
-  }
+  } 
 
   state[currentStep] += num
   updateScreen(state[currentStep])
 }
 
+const addDecimal = () => {
+  let currentStep
+
+  // if the 'operator' has been set we know we're dealing with the second number
+  if (state['operator'] === '') {
+    currentStep = 'firstNumber'
+  } else {
+    currentStep = 'secondNumber'
+  } 
+
+  if (state[currentStep].indexOf('.') == -1) {
+    (state[currentStep]) += ('.')
+  }  else { 
+    buttonDecimal.disabled = true
```

Comment:
`buttonDecimal` is undefined here.

Reply:
Assume for the sake of argument that `buttonDecimal` isn't undefined and this disables the "=" button.

What would happen if we were to try to add a decimal to _both_ the first and second numbers?

If a user tries to add a decimal more than once they're effectively locked out; if they try to add it to the second number it'll still be disabled.